custom_components/waste_collection_schedule/waste_collection_schedule/source/zys_harmonogram_pl.py

```diff
@@ -0,0 +1,123 @@
+####Mod by SEMATpl - semat.pl - based on sepan_remondis_pl ###
+import datetime
+import json
+import logging
+import xml.etree.ElementTree
+
+import requests
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+
+TITLE = "Kleszczewo/Kostrzyn"
+DESCRIPTION = "Source for Kleszczewo/Kostrzyn commune garbage collection"
+URL = "https://www.puk-zys.pl/index.php/harmonogram-wywozow-2024-r/" #Only Kleszczewo and Kostrzyn
+TEST_CASES = {
+    "Street Name": {
+        "city": "Komorniki",
+        "street_name": "Komorniki",
+        "street_number": "93/2",
+        "commune_name": "Komorniki"
+    },
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+API_URL = "https://zys-harmonogram.smok.net.pl/{}/{}"
+
+NAME_MAP = {
+    1: "Zmieszane odpady komunalne",
+    2: "Papier",
+    3: "Metale i tworzywa sztuczne",
+    4: "Szkło",
+    5: "Bioodpady",
+    6: "Drzewka świąteczne",
+    7: "Odpady wystawkowe",
+}
+
+ICON_MAP = {
+    1: "mdi:trash-can",
+    2: "mdi:recycle",
+    3: "mdi:recycle",
+    4: "mdi:recycle",
+    5: "mdi:recycle",
+    6: "mdi:recycle",
+    7: "mdi:trash-can",
+}
+
+
+class Source:
+    def __init__(self, city, street_name, street_number, commune_name):
+        self._city = city.upper()
+        self._street_name = street_name.upper()
+        self._street_number = street_number.upper()
+        self._commune_name = commune_name.lower()
+
+
+    def fetch(self):
+        try:
+            return self.get_data(API_URL.format(self._commune_name,datetime.datetime.now().year))
+        except Exception:
+            _LOGGER.debug(
+                f"fetch failed for source {TITLE}: trying different API_URL ..."
+            )
+            return self.get_data(API_URL.format(""))
+
+    def get_data(self, api_url):
+        r = requests.get(f"{api_url}/addresses/cities")
+        r.raise_for_status()
+        city_id = 0
+        cities = json.loads(r.text)
+        for item in cities:
+            if item["value"] == self._city:
+                city_id = item["id"]
+        if city_id == 0:
+            raise Exception("city not found")
+
+        r = requests.get(f"{api_url}/addresses/streets/{city_id}")
+        r.raise_for_status()
+        street_id = 0
+        streets = json.loads(r.text)
+        for item in streets:
+            if item["value"] == self._street_name:
+                street_id = item["id"]
+        if street_id == 0:
+            raise Exception("street not found")
+
+        r = requests.get(f"{api_url}/addresses/numbers/{city_id}/{street_id}")
+        r.raise_for_status()
+        number_id = 0
+        numbers = json.loads(r.text)
+        for item in numbers:
+            if item["value"] == self._street_number:
+                number_id = item["id"]
+        if number_id == 0:
+            raise Exception("number not found")
+
+        r = requests.get(f"{api_url}/reports?type=html&id={number_id}")
```

Comment:
I'd prefer to use the params argument (`r = requests.get(f"{api_url}/reports", params={"type": "html","id": number_id})`

Reply:
I agree with you:)
I fixed it in the latest commit.